pkg/placement/placement-resolver.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placement
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kubestellar/kubestellar/api/edge/v1alpha1"
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+// PlacementResolver abstracts the functionality needed from a placement resolver,
+// to simplify its usage and keep implementation details inaccessible.
```

Comment:
The comment on this type needs to give a brief outline of what the functionality is; the existing comment is more of a report on the reason this type exists --- which is not bad, but the reader needs to also get a brief description of what this thing is.

The comments on the methods below use mixed terminology. Some speak of "the cache", others speak of "the maintained placement decision".

If I understand correctly, a brief accurate description would be something like the following.

> A PlacementResolver holds a collection of placement decisions. The collection is indexed by `placementKey string`, the resolver does not care what the strings are. Decisions can be updated, externalized, and compared with an external representation.

Reply:
What do you think about

>// A PlacementResolver holds a collection of placement resolutions.
// The collection is indexed by placementKey string, the resolver does not
// care what the strings are, but a string should be a placement's unique
// identifier. The resolution for a given key can be updated,
// exported and compared to the PlacementDecision representation.

Reply:
@MikeSpreitzer 
All comments until this point have been addressed. The first batch in the commits referenced in a similar comment above, and this 2nd batch in commits: [addressed mike's 2nd batch of comments in PR 1681, until comment refrencing this commit](https://github.com/kubestellar/kubestellar/pull/1681/commits/ee81ae6b654d52fd635bd30ecd420fa8318c3bce) and [updated resolution comparison functions documentation](https://github.com/kubestellar/kubestellar/pull/1681/commits/55cb1cb811ea64a20c74c7e3b93e206cd89ef1c6).

Outdated comments below have also been addressed.